skaraMirror.sh

```diff
@@ -192,20 +192,15 @@ TAG_SEARCH="jdk-${GITHUB_REPO//[!0-9]/}*+*"
 
 # JDK11+ tag sorting:
 # We use sort and tail to choose the latest tag in case more than one refers the same commit.
-# Versions tags are formatted: jdk-V[.W[.X]]+B; with V, W, X, B being numeric.
-# Transform "-" to "." in tag so we can sort as: "jdk.V[.W[.X]]+B"
-#
-# Example: jdk-11.0.2+2, jdk-11.0.1+3, jdk-11.0.2+1
-# Sort by "B": jdk-11.0.2+1, jdk-11.0.2+2, jdk-11.0.1+3
-# Then, "Stable" sort by V.W.X: jdk-11.0.1+3, jdk-11.0.2+1, jdk-11.0.2+2
-# Latest = jdk-11.0.2+2
-#
+# Versions tags are formatted: jdk-V[.W[.X[.P]]]+B; with V, W, X, P, B being numeric.
+# Transform "-" to "." in tag so we can sort as: "jdk.V[.W[.X[.P]]]+B"
+# Transform "+" to ".0.+" during the sort so that .P (patch) is defaulted to "0" for those
+# that don't have one, and the trailing "." to terminate the 5th field from the +
 # First, sort on build number (B):
-jdk11plus_tag_sort1="sort -t+ -k2n"
-# Second, (stable) sort on (V), (W), (X):
-jdk11plus_tag_sort2="sort -t. -k2n -k3n -k4n -s"
-jdk11plus_sort_tags_cmd="grep -v _adopt | sed 's/jdk-/jdk./g' | $jdk11plus_tag_sort1 | $jdk11plus_tag_sort2 | sed 's/jdk./jdk-/g'"
-
+jdk11plus_tag_sort1="sort -t+ -k2,2n"
+# Second, (stable) sort on (V), (W), (X), (P): P(Patch) is optional and defaulted to "0"
+jdk11plus_tag_sort2="sort -t. -k2,2n -k3,3n -k4,4n -k5,5n"
```

Comment:
This sort command isn't equivalent to the old, stable approach; the final criteria needs to be the first column (the number inserted by `nl`).

See also https://github.com/adoptium/temurin-build/pull/2653, which might help here too (although, in some cases, more than one tag is expected from `jdk11plus_sort_tags_cmd`).

Reply:
@keithc-ca I might be wrong, but I thought using the "nl" method did make it stable by leveraging "last-resort" comparison, which is what -s disables:

       Finally, as a last resort when all keys compare equal (or if no	order-
       ing  options  were  specified  at all), sort compares the lines byte by
       byte in machine collating sequence.  The last resort comparison	honors
       the -r global option.  The -s (stable) option disables this last-resort
       comparison so that lines in which all fields compare equal are left  in
       their  original	relative  order.   If  no fields or global options are
       specified, -s has no effect.

"nl" line numbers are added prior to sort2, thus if during sort2 all the key fields are identical, then "last-resort" comparison of the whole line thus beginning with the "nl" line number does the final sort, thus keeping it stable.

Reply:
Yes, the last resort comparison will involve whole lines, but it won't treat the first column as a number and so would order, for example, '28' after '100'. (see also https://github.com/adoptium/temurin-build/pull/2653#issuecomment-858583760)